app/controllers/polls_controller.rb

```diff
@@ -23,8 +28,54 @@ def new
 
   # GET /polls/1/edit
   def edit
-    if current_user.is_student
-      redirect_to lecture_poll_path(@lecture, @poll)
+    if @is_student
+      if !@poll.is_active
+        redirect_to lecture_polls_path(@lecture), notice: "This poll is closed."
+      else
+        get_users_answers
+        render :answer
+      end
+    end
+  end
+
+  # GET /polls/1/stop
+  def stop_start
+    if @poll.is_active
+      if @poll.update(is_active: false)
+        redirect_to lecture_poll_path(@lecture, @poll), notice: "You stopped the poll!"
+      else
+        redirect_to lecture_poll_path(@lecture, @poll), notice: "Stopping the poll did not work :("
+      end
+    else
+      if @poll.update(is_active: true)
+        redirect_to lecture_poll_path(@lecture, @poll), notice: "You started the poll!"
+      else
+        redirect_to lecture_poll_path(@lecture, @poll), notice: "Starting the poll did not work :("
+      end
+    end
+  end
+
+  # POST /polls/id/save_answers
+  def save_answers
+    answer_params
+    current_poll_answers = params[:answers]
+    poll = Poll.find(params[:id])
+    if !poll.is_active
+        redirect_to lecture_polls_path(@lecture), notice: "This poll is closed, you cannot answer it."
+    else
+      # delete answers from student to poll
```

Comment:
every code block you can comment you can also extract!

Reply:
Well, that's only describing the following line. Instead of extracting that I removed the comment, because it is kind of obvious what we are doing ^.^